### DISCUSSION TITLE
how to add custom mutator reference?

### DISCUSSION THREAD

**Discussion by ManuRodgers** [pmndrs/zustand#1304] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/33333844/191624803-12361aab-5afa-4be7-b4ba-716db849a3a0.png)

**Comment by dai-shi** [comment, marked as answer]
What's your goal? Can you create an example with typescript playground? Can start from here: https://tsplay.dev/mbn5Pm

Are you trying to change the store type?
https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#middleware-that-changes-the-store-type
